ui/app/routes/vault/cluster/secrets/backend/secret-edit.js

```diff
@@ -213,10 +212,10 @@ export default Route.extend(UnloadModelRoute, {
     return secretModel;
   },
 
-  async model(params) {
+  async model(params, transition) {
     let secret = this.secretParam();
     let backend = this.enginePathParam();
-    let modelType = this.modelType(backend, secret);
+    let modelType = this.modelType(backend, secret, transition);
```

Comment:
I don't love needing to pass around the transition for the type. Could we instead get the `itemType` param here in the model hook (and the other spots where the type is passed in) and then pass it to modelType as a string?

Reply:
I agree it's not my favorite either. On line 71 the `buildModel` method is called in the `beforeModel` hook which then calls `modelType`. Since only the transition is available in `beforeModel` that's what I chose to pass along to keep it consistent.

I guess another reason I elected to pass along the transition was if we pull out `itemType` then we have an arg to `modelType` that is specific to keymgmt whereas with the transition there is access to the route params and query params if a case like this comes up again in the future for a different backend.

Reply:
If we need the param in a future secret engine, I would suggest we reuse the "itemType" param for that secret engine as well. I think the core issue is that in principle, `modelType` method shouldn't need transition, but maybe we pass in an object "options" that can be extended in the future. That way we can pass in all the queryparams from the transition in the caller of `modelType`.

Reply:
Ok that sounds good to me I will update it to pass an `options` object that for now will have the `queryParams` from the transition.